src/tools/prefs.win.cpp

```diff
@@ -134,7 +149,15 @@ static bool load_prefs(std::istream& in) {
 		std::string key = line.substr(0, key_end + 1), val = line.substr(val_beg);
 		if(val == "true") temp_prefs[key] = true;
 		else if(val == "false") temp_prefs[key] = false;
-		else if(val[0] == '[') {
+		else if(val[0] == '"') {
+			// Dastardly trick to make TinyXML encode and decode string preferences
```

Comment:
I distinctly recall a function existing in oboe to handle exactly this case…

(looks it up)

There it is in fileio.cpp: `read_maybe_quoted_string` and `maybe_quote_string`. Is there a good reason not to use these? Admittedly, they only escape the quotes themselves (and the escape character and some whitespace), whereas your version probably escapes a bunch of things, but I'm not sure if there's any need to escape those other things.

Reply:
If I'd known about that, I would have used it. Definitely better.